### PR TITLE
Allow for analysis under rebar3_hank

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ${{matrix.os}}
     strategy:
       matrix:
-        otp_vsn: [20, 21, 22, 23, 24]
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:16]: `rebar3_hank` is 21+

File: test/test_app_SUITE.erl [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 -export([all/0, test_app/1]).
 
+-hank([{unnecessary_function_arguments, [test_app/1]}]).
```
Comment (elbrujohalcon) [inline test/test_app_SUITE.erl:5]: This should not be necessary. Just adding `-behavior(ct_suite).` to the module should be enough.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: test/test_app_SUITE.erl (hunk 2 above)
Comment (paulo-ferraz-oliveira) [reply test/test_app_SUITE.erl:5]: This is under CI; that behaviour is not 21-compatible.

Comment (elbrujohalcon) [reply test/test_app_SUITE.erl:5]: What we did (at NextRoll) was to, either…
1. stop checking the project with <23 OTP versions on CI
2. remove `warnings_as_errors` until we could do that ☝🏻

Comment (elbrujohalcon) [reply test/test_app_SUITE.erl:5]: I mean… the addition of `-behavior(ct_suite).` _just_ triggers a warning on versions smaller than OTP23, it doesn't _brake_ the code.

Comment (elbrujohalcon) [reply test/test_app_SUITE.erl:5]: Worst case scenario, I'm open to something like having a define in rebar.config's `erl_opts` and then…

```erlang
-ifdef(OTP_VERSION_ABOVE_23).
-behavior(ct_suite).
-endif.
```

But my favorite approach would be to just removing CI support for OTP <23.

Comment (paulo-ferraz-oliveira) [reply test/test_app_SUITE.erl]: Pushed.

File: rebar.config [hunk 3]
```diff
@@ -1,3 +1,5 @@
+{minimum_otp_vsn, "21"}.
```
Comment (paulo-ferraz-oliveira) [inline rebar.config:1]: Because we're using macro `OTP_RELEASE`.

File: test/test_app_SUITE.erl [hunk 4]
```diff
@@ -2,6 +2,14 @@
 
 -export([all/0, test_app/1]).
 
+-if(?OTP_RELEASE >= 24).
+-behaviour(ct_suite).
+-else.
+-hank([{unnecessary_function_arguments, [
+            test_app/1
+       ]}]).
+-endif.
```
Comment (elbrujohalcon) [inline test/test_app_SUITE.erl:11]: Hank does handle ct suites prior to `24` (when `ct_suite` behavior didn't exist).
I guess the issue is that it's only for _some_ variants of `OTP23` (i.e. it was introduced in… say… `OTP23.2` or something).
In any case… if we _really really_ need to preserve the `-hank(…)` part, it should only be there for `?OTP_RELEASE == 23`.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]

File: test/test_app_SUITE.erl (hunk 4 above)
Comment (paulo-ferraz-oliveira) [reply test/test_app_SUITE.erl]: Handled.

**Review by elbrujohalcon — APPROVED** [review]

**Pull request merged**